qiskit/circuit/controlflow/switch_case.py

```diff
@@ -176,3 +179,221 @@ def c_if(self, classical, val):
             "SwitchCaseOp cannot be classically controlled through Instruction.c_if. "
             "Please nest it in an IfElseOp instead."
         )
+
+
+class SwitchCasePlaceholder(InstructionPlaceholder):
+    """A placeholder instruction to use in control-flow context managers, when calculating the
+    number of resources this instruction should block is deferred until the construction of the
+    outer loop.
+
+    This generally should not be instantiated manually; only :obj:`.SwitchContext` should do it when
+    it needs to defer creation of the concrete instruction.
+
+    .. warning::
+
+        This is an internal interface and no part of it should be relied upon outside of Qiskit
+        Terra.
+    """
+
+    def __init__(
+        self,
+        target: Union[Clbit, ClassicalRegister],
+        cases: List[Tuple[Any, ControlFlowBuilderBlock]],
+        *,
+        label: Optional[str] = None,
+    ):
+        self.__target = target
+        self.__cases = cases
+        self.__resources = self._placeholder_resources()
+        super().__init__(
+            "switch_case",
+            len(self.__resources.qubits),
+            len(self.__resources.clbits),
+            [],
+            label=label,
+        )
+
+    def _placeholder_resources(self):
```

Comment:
Could we use a different method name for this? It took me an embarrassingly long amount of time to figure out why L207 didn't error until I realized this was called `_placeholder_resources` which wasn't the same as the public (and abc required) `placeholder_resources` below.

Reply:
sure - I'll change it here and if `IfContext` to to `_calculate_placeholder_resources` or something

Reply:
Done in 41d05b624.